src/runtime_src/doc/toc/security.rst

```diff
@@ -0,0 +1,120 @@
+to ensure that the user design physically confines itself to DR partition and does not attempt
+to overwrite portions of the Shell.
+
+
+Firewall
+========
+
+Alveo hardware design uses standard AXI bus. As shown in the figure the control path uses AXI-Lite
+and data path uses AXI4 full. Specialized hardware element called AXI Firewall monitors all transactions
+going across the bus into the bun-trusted DFX partition. It is possible that one or more AXI slave in the DFX
+partition is not fully AXI-compliant or deadlocks/stalls/hangs during operation. At this point to protect
+the Shell, AXI Firewall trips -- it starts completing AXI transactions on behalf of the slave so the master
+and the specific AXI bus is not impacted. The AXI Firewall starts completing all transactions on behalf of
+the slave while also notifying the mgmt driver about the trip. The mgmt driver then starts taking recovery
+action.
+
+Deployment Models and Trust Roles
+=================================
+
+Baremetal
+---------
+
+In Baremetal deployment model, both physical functions are visible to the end user. End users interact
+with both xclmgmt and xocl drivers directly. The system administrator trusts both drivers. End users
+have the privilege to load xclbins which should be signed for most security. Ideally in an enterprise
+the system administrator should sign the xclbins with an enterprise key which is registered with the
+system key-ring. In this case xclmgmt driver would only permit signed xclbins to be loaded. This will
+ensure that only known good xclbins are loaded by end users.
+
+Certain operations like resetting the board and upgrading the flash image on PROM (from which the shell is
+loaded on system boot) require root privileges and are effected by xclmgmt driver.
+
+Pass-through Virtualization
+---------------------------
+
+In Pass-through Virtualization deployment model, management physical function is only visible to the host
+but user physical function is visible to the guest VM. Users in guest VM cannot perform any privileged
+operation like updating flash image or device reset. Since xclbin downloads are done by xclmgmt driver
+xclbins are passed on to the host via a plugin based MPD/MSD defined in :ref:`mailbox.main.rst`. Host can
+add any extra checks necessary to validate xclbins received from guest VM. This deployment model is ideal
+for public cloud where host does not trust the guest VM. This is the prevalent deployment model for FaaS
+operators.
+
+Signing of Xclbins
+==================
+
+xclbin signing process is similar to signing of Linux kernel modules. xclbins can be signed by XRT utility,
+``xclbinutil``. The signing adds a PKCS7 signature at the end of xclbin. The signing certificate is then
+registered with system key-ring. When xclbin is provided to xclmgmt driver it will refuse to load xclbin
+if xclbin signature valiation fails against the certificates in its key ring.
+
```

Comment:
There is another security level (level 1) at which ".xilinx_fpga_xclbin_keys" key-ring (instead of system key-ring, which is level 2) will be used. Any one with root privilege can add keys to this key-ring at run time. No need to go through UEFI.
If secure boot is used (driver automatically detects it), level 1 is not available and level 2 will be activated automatically and system key-ring will be used. Otherwise, sys admin can promote secure level from 0 (nothing) to 1 and/or to 2 at run time using xbmgmt.

Reply:
Added description of levels. Please review.